### PR TITLE
Fix quoting of paths in various places

### PR DESCRIPTION
**Pull request by gwenzek**
Initially for #1600, but the LocalExecutor also had some minor bugs.
I added some more shlex.quote in the code.

New failing tests:
https://app.circleci.com/pipelines/github/facebookincubator/submitit/151/workflows/46d4118f-9262-49b4-936b-08b5ce9d830a/jobs/715/tests

Fix tests:
https://app.circleci.com/pipelines/github/facebookincubator/submitit/152/workflows/25407c90-c3bd-421f-9144-4b1810269d6f/jobs/723/tests

### CODE REVIEW

File: submitit/slurm/test_slurm.py [hunk 1]
```diff
@@ -343,3 +343,30 @@ def test_slurm_invalid_parse() -> None:
 def test_slurm_missing_node_list() -> None:
     with with_slurm_job_nodelist("") as env:
         assert [env.hostname] == env.hostnames
+
+
+def test_slurm_weird_dir(tmp_path: Path, weird_dir: str) -> None:
+    if "\n" in weird_dir:
+        pytest.skip("test doesn't support newline in 'weird_dir'")
```
Comment (jrapin) [inline submitit/slurm/test_slurm.py:350]: is it supported by slurm? if not then raise there maybe?

File: submitit/conftest.py [hunk 2]
```diff
@@ -8,3 +8,8 @@
 @pytest.fixture()
 def executor(tmp_path: Path) -> LocalExecutor:
     return LocalExecutor(tmp_path)
+
+
+@pytest.fixture(params=["a_0", "a 0", 'a"=0"', "a'; echo foo", r"a\=0", r"a\=", "a\n0"])
+def weird_dir(request) -> str:
+    return request.param
```
Comment (jrapin) [inline submitit/conftest.py:15]: you could send a temporary path instead directly :D

**Review by jrapin — APPROVED** [review]

File: submitit/conftest.py (hunk 2 above)
Comment (gwenzek) [reply submitit/conftest.py:15]: good idea ! done

File: submitit/slurm/test_slurm.py (hunk 1 above)
Comment (gwenzek) [reply submitit/slurm/test_slurm.py:350]: I believe that newline in filename is supported by sbatch but not by my crappy sbatch parsing test code. Can I change the message to make this more clear ?

**Pull request merged**